### PR TITLE
🛠️  Add go workspace 

### PR DESCRIPTION
**Pull request by moogacs**
### What's being changed:
this PR creates go workspace so that we can sync all submodules with main module dependancies, once we update any dependancy in the  main module , we just need to run `go mod tidy && go work sync`

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Comment by moogacs** [comment]
hmm, i see some failure regarding using the go.work file, need to  re-check

File: go.mod [hunk 1]
```diff
@@ -55,7 +55,7 @@ require (
 	github.com/hashicorp/raft v1.5.0
 	github.com/hashicorp/raft-boltdb/v2 v2.2.2
 	github.com/pbnjay/memory v0.0.0-20210728143218-7b4eea64cf58
-	github.com/tailor-inc/graphql v0.2.1
+	github.com/tailor-inc/graphql v0.4.1
```
[the remark this answers is not in the document]
Comment (moogacs) [reply go.mod:58]: i can pin it to what it was to be on the safe zone

[the remark this answers is not in the document]
Comment (dirkkul) [reply go.mod:58]: I can only remember that we switched from the original vendor to this one - you could run the go client integration tests with the updated server to be sure that it is not a problem

[the remark this answers is not in the document]
Comment (moogacs) [reply go.mod:61]: i add a replace with `v0.2.1`  if we wanna proceed without upgrading

File: .dockerignore [hunk 2]
```diff
@@ -37,3 +37,7 @@ var/weaviate/
 
 # ignore archives, they are most likely part of some debugging and never part of actual code
 *.tar.gz
+
+# go workspace files
+go.work
+go.work.sum
```
Comment (antas-marcin) [inline .dockerignore:43]: please add an empty line here

**Review by reyreaud-l — DISMISSED** [review]

**Comment by moogacs** [comment]
replaced by https://github.com/weaviate/weaviate/pull/5159

**Pull request closed without merging**